src/main/java/io/debezium/connector/yugabytedb/connection/OpId.java

```diff
@@ -163,8 +163,17 @@ public CdcSdkCheckpoint toCdcSdkCheckpoint() {
      * the corresponding values in {@link CdcSdkCheckpoint}
      */
     public boolean isLesserThanOrEqualTo(CdcSdkCheckpoint checkpoint) {
-        return (checkpoint != null && this.term <= checkpoint.getTerm()
-                && this.index <= checkpoint.getIndex() && this.time <= checkpoint.getTime());
+        if (checkpoint == null) {
+            return false;
+        }
+
+        if (this.term < checkpoint.getTerm() || this.index < checkpoint.getIndex() || this.time < checkpoint.getTime()) {
```

Comment:
Should't we have `AND` in place of `OR` between the clauses here?

Reply:
Suppose we have a case where we received the finalOpId as `1.2.[].0.100` and we genuinely do not have anything further in the WAL, not even a `NO_OP`, in that case our time in the response checkpoint will keep on increasing(`1.2.[].0.101`, `1.2.[].0.102`, `1.2.[].0.103` and so on) without the `term` or `index` moving ahead - we would want to increment the advanced checkpoint even when the only increment is in time.

If we add an `AND` here instead of `OR`, we will end up failing the check and won't increment the checkpoint contrary to what we are trying to achieve here.